mypy/report.py

```diff
@@ -156,14 +160,66 @@ def on_file(self,
         visitor = stats.StatisticsVisitor(inferred=True, filename=tree.fullname(),
                                           typemap=type_map, all_nodes=True)
         tree.accept(visitor)
+        self.any_types_counter[tree.fullname()] = visitor.type_of_any_counter
         num_unanalyzed_lines = list(visitor.line_map.values()).count(stats.TYPE_UNANALYZED)
         # count each line of dead code as one expression of type "Any"
-        num_any = visitor.num_any + num_unanalyzed_lines
-        num_total = visitor.num_imprecise + visitor.num_precise + num_any
+        num_any = visitor.num_any_exprs + num_unanalyzed_lines
+        num_total = visitor.num_imprecise_exprs + visitor.num_precise_exprs + num_any
         if num_total > 0:
             self.counts[tree.fullname()] = (num_any, num_total)
 
     def on_finish(self) -> None:
+        self._report_any_exprs()
+        self._report_types_of_anys()
+
+    def _report_types_of_anys(self) -> None:
+        # On the following line, mypy complains about implicit generic Any and wants us to use
+        # typing.Counter() instead of collections.Counter().
+        # We cannot use typing.Counter() because it doesn't work with python 3.5 and older.
+        total_counter = collections.Counter()  # type: ignore
```

Comment:
Instead, give this `# type: typing.Counter[str]` (or whatever the item type is, exactly).